apps/common/hooks/useBalances.tsx

```diff
@@ -63,188 +64,210 @@ const		defaultStatus = {
+			result = {...result, ...callResult};
+		} else {
+			console.error(_error);
+		}
+	}
+	return [result, undefined];
+}
+
+
 /* 🔵 - Yearn Finance ******************************************************
 ** This hook can be used to fetch balance information for any ERC20 tokens.
 **************************************************************************/
 export function	useBalances(props?: TUseBalancesReq): TUseBalancesRes {
+	const	workerRef = useRef<Worker>();
 	const	{address: web3Address, isActive, provider} = useWeb3();
 	const	{chainID: web3ChainID} = useChainID();
+	const	{onLoadStart, onLoadDone} = useUI();
 	const	[nonce, set_nonce] = useState(0);
 	const	[status, set_status] = useState<TDefaultStatus>(defaultStatus);
 	const	[error, set_error] = useState<Error | undefined>(undefined);
 	const	[balances, set_balances] = useState<TNDict<TDict<TBalanceData>>>({});
 	const	data = useRef<TNDict<TDataRef>>({1: {nonce: 0, address: toAddress(), balances: {}}});
-	const	interval = useRef<NodeJS.Timer>();
-	const	effectDependencies = props?.effectDependencies || [];
+	const	stringifiedTokens = useMemo((): string => JSON.stringify(props?.tokens || []), [props?.tokens]);
 
-	/* 🔵 - Yearn Finance ******************************************************
-	** When this hook is called, it will fetch the informations for the
-	** specified list of tokens. If no props are specified, the default values
-	** will be used.
-	**************************************************************************/
-	const stringifiedTokens = useMemo((): string => JSON.stringify(props?.tokens || []), [props?.tokens]);
-
-	async function performCall(
-		ethcallProvider: Provider,
-		calls: Call[],
-		tokens: TUseBalancesTokens[]
-	): Promise<[TDict<TBalanceData>, Error | undefined]> {
-		const	_data: TDict<TBalanceData> = {};
-		const	results = await ethcallProvider.tryAll(calls);
-
-		let		rIndex = 0;
-		for (const element of tokens) {
-			const	{token} = element;
-			const	balanceOf = results[rIndex++] as BigNumber;
-			const	decimals = results[rIndex++] as number;
-			const	rawPrice = formatBN(props?.prices?.[toAddress(token)]);
-			let symbol = results[rIndex++] as string;
-
-			if (toAddress(token) === ETH_TOKEN_ADDRESS) {
-				symbol = 'ETH';
-			}
-			_data[toAddress(token)] = {
-				decimals: Number(decimals),
-				symbol: symbol,
-				raw: balanceOf,
-				rawPrice,
-				normalized: format.toNormalizedValue(balanceOf, Number(decimals)),
-				normalizedPrice: format.toNormalizedValue(rawPrice, 6),
-				normalizedValue: (format.toNormalizedValue(balanceOf, Number(decimals)) * format.toNormalizedValue(rawPrice, 6))
+	const	updateBalancesFromWorker = useCallback((
+		workerForChainID: number,
+		newRawData: TDict<TBalanceData>,
+		err?: Error
+	): TDict<TBalanceData> => {
+		console.warn(newRawData, err);
+		if (toAddress(web3Address as string) !== data?.current?.[workerForChainID]?.address) {
```

Comment:
I'd prefer to update the `toAddress` to handle the `null` rather than doing this casting